editor/src/components/editor/store/store-monitor.ts

```diff
@@ -0,0 +1,23 @@
+import type { ElementPath } from 'utopia-shared/src/types'
+import { Substores, useEditorState } from './store-hook'
+import { mapDropNulls } from '../../../core/shared/array-utils'
+import { withUnderlyingTarget } from './editor-state'
+
+export function useMonitorChangesToElements(elementPaths: Array<ElementPath>): void {
+  useEditorState(
+    Substores.projectContents,
+    (store) => {
+      return mapDropNulls((elementPath) => {
+        return withUnderlyingTarget(
+          elementPath,
+          store.editor.projectContents,
+          null,
+          (_, element) => {
+            return element
+          },
+        )
+      }, elementPaths)
+    },
+    'useMonitorChangesToElements elements',
+  )
+}
```

Comment:
This means that the rerender is dependent on project code change, but the positioning of a grid can change without that if it depends on the state of the component, or on loading external data (e.g. what if you select a grid before the metaobjects which fill it are loaded?), maybe even on remix route

Reply:
I've implemented an alternative approach for triggering in 952e26eff. But also added a `useKeepReferenceEqualityIfPossible` call around the result of both `useGridData` and `useGridMeasurementHelperData` so that we don't re-render the controls themselves unnecessarily.